pkg/reconciler/apis/apibinding/apibinding_reconcile.go

```diff
@@ -399,12 +403,30 @@ func (c *controller) reconcileBinding(ctx context.Context, apiBinding *apisv1alp
 				"Waiting for API(s) to be established: %s", strings.Join(needToWaitForRequeueWhenEstablished, ", "),
 			)
 		}
+		return nil
 	} else {
 		conditions.MarkTrue(apiBinding, apisv1alpha1.InitialBindingCompleted)
 		conditions.MarkTrue(apiBinding, apisv1alpha1.BindingUpToDate)
 		apiBinding.Status.Phase = apisv1alpha1.APIBindingPhaseBound
 	}
 
+	// Once the APIBinding is bound and initialized,, update the APIExport's VirtualWorkspaceURL
+	newAPIExport := apiExport.DeepCopy()
```

Comment:
This is in draft because I think this behavior's not complete. The APIExport controller adds event handlers to the ClusterWorkspaceShard informer to enqueue all APIExports within a given ClusterWorkspaceShard, presumably to update the list of virtual workspace URLs.

It seems clunky to me to do the same with APIBindings, because we'd be enqueueing an APIBinding to update an APIExport. There's also the possibility that the APIBinding controller creates the _first_ virtualWorkspaceURL and the APIExport controller handles updates, but that spreads the responsibility across too many actors.

Open to suggestions here.

Reply:
> It seems clunky to me to do the same with APIBindings, because we'd be enqueueing an APIBinding to update an APIExport.

This is fairly typical behavior. I would go this route, I think